File: 326. Power of Three/isPowerOfThree.py
class Solution:
    def isPowerOfThree(self, n: int):
        if n == 3 or n == 1:
            return True
        if n < 3 or (n % 3) != 0:
            return False
        return self.isPowerOfThree(n/3)

sol = Solution()
print(sol.isPowerOfThree(27))

File: 326. Power of Three/is_power_of_three.rs
impl Solution {

    // O(logn) time,
    // O(logn) space,
    // Approach: recursion
    pub fn is_power_of_three(n: i32) -> bool {
        if n == 1 {
            return true;
        }
        
        if n <= 0 || n % 3 != 0 {
            return false;
        }
        
        Solution::is_power_of_three(n/3)
    }

    // O(logn) time,
    // O(1) space,
    // Approach: recursion
    pub fn is_power_of_three2(mut n: i32) -> bool {
        
        while n > 1 {
            if n % 3 != 0 {
                return false;
            }

            n /= 3;
        }

        n == 1
    }
}